README.md

```diff
@@ -1,54 +1,73 @@
+![Figure 1. Data Analysis Flow Chart](doc/flow_chart.jpg)
+
+1.  **src/download_data.py** - A first script that downloads the data from the UCI repository and saves it as a csv
+2.  **src/split_data.py** - A second script that takes in the data generated from first script and save the partitioned data (train data and test data) in csv format.
+3.  **src/eda.py** - A third script that reads the train data, creates plots for the provided training data file for analysis.
+4.  **src/preprocess_model_selection.py** - A fourth script that reads the train data and performs preprocessing, builds a model and render the model file and cross validation result.
+5.  **src/predict.py** - A fifth script that takes in the model file generated from the fourth script as well as test data generated from the second script and generate the test results.
+6.  **doc/countraceptive_method_predictor_report.Rmd** - A sixth script that presents the key exploratory data analysis from the third script as well as the test result summaries from the fifth script.
 
 ## Usage
 
-To replicate the analysis, clone this GitHub repository, install the depndencies listede below, and run the following command at the commandline / terminal from the root directory of this project:
+To replicate the analysis, clone this GitHub repository, install the dependencies listed below, and run the following command at the command line / terminal from the root directory of this project:
 
     python src/download_data.py --url=https://archive.ics.uci.edu/ml/machine-learning-databases/cmc/cmc.data --out_file=data/raw/contraceptive.csv
-    jupyter lab src/eda.ipynb
+
+
+    python src/eda.py --train_dir=data/processed/train.csv --out_dir=./results
 
 ## Initial EDA
 
-At this intial stage we have done an EDA of the dataset and have found that 
+At this initial stage we have done an EDA of the dataset and have found that
 
-- Majority of the women(~43% in the given dataset) are not using any contraceptive method
-- Majority of the observations are  of the 'wife age' between 22-36years
-- Most of them have 2-3 children
-- Women have high level of education, Husbands too have high level of education.
-- Most of women are not working
-- Women have high standard of living albeit with no media exposure
+-   Majority of the women(\~43% in the given dataset) are not using any contraceptive method
+-   Majority of the observations are of the 'wife age' between 22-36years
+-   Most of them have 2-3 children
+-   Women have high level of education, Husbands too have high level of education.
+-   Most of women are not working
+-   Women have high standard of living albeit with no media exposure
 
 ## Dependencies
 
-- Python 3.7.3 and Python packages:
-    - docopt==0.6.2
-    - pandas==1.3.4
-    - altair==4.1.0
-    - numpy==1.21.4
-    - scikit-learn==1.0.1
+-   Python 3.7.3 and Python packages:
```

Comment:
Not sure if we need this considering we are proving an environment in our repository. We can rather show how to create environment using the command as mentioned in the previous comment.